### PR TITLE
Add Warnings for deprecated attribute types

### PR DESCRIPTION
**Pull request by m-luo16**

### CODE REVIEW

File: misk-launchdarkly-core/src/main/kotlin/misk/feature/launchdarkly/LaunchDarklyFeatureFlags.kt [hunk 1]
```diff
@@ -132,13 +142,25 @@ class LaunchDarklyFeatureFlags @Inject constructor(
...
+      attributes.number!!.forEach { (k, v) ->
+        if (v is Long) {
+          logger.info { "Please use an Int for $k" }
+        }
+        if (v is Float) {
+          logger.info { "Please use a Double for $k" }
+        }
+        builder.privateCustom(k, v)
+      }
+      if (attributes.anonymous) {
```
Comment (escardin) [inline misk-launchdarkly-core/src/main/kotlin/misk/feature/launchdarkly/LaunchDarklyFeatureFlags.kt:151]: I think this got pushed in with the number processing?

File: misk-launchdarkly-core/src/main/kotlin/misk/feature/launchdarkly/LaunchDarklyFeatureFlags.kt [hunk 2]
```diff
@@ -132,13 +142,25 @@ class LaunchDarklyFeatureFlags @Inject constructor(
...
-    }
-    if (attributes.anonymous) {
-      // This prevents the user from being stored in the LaunchDarkly dashboard, see
-      // https://docs.launchdarkly.com/docs/anonymous-users
-      builder.anonymous(true)
+      attributes.number!!.forEach { (k, v) ->
+        if (v is Long) {
+          logger.info { "Please use an Int for $k" }
+        }
+        if (v is Float) {
```
Comment (escardin) [inline misk-launchdarkly-core/src/main/kotlin/misk/feature/launchdarkly/LaunchDarklyFeatureFlags.kt]: It's reasonable to convert a float to a double

**Review by escardin — APPROVED** [review]
Approved with comments

File: misk-launchdarkly-core/src/main/kotlin/misk/feature/launchdarkly/LaunchDarklyFeatureFlags.kt [hunk 3]
```diff
@@ -132,13 +142,22 @@ class LaunchDarklyFeatureFlags @Inject constructor(
...
     if (attributes.number != null) {
-      attributes.number!!.forEach { (k, v) -> builder.privateCustom(k, v) }
-    }
-    if (attributes.anonymous) {
-      // This prevents the user from being stored in the LaunchDarkly dashboard, see
-      // https://docs.launchdarkly.com/docs/anonymous-users
-      builder.anonymous(true)
+      attributes.number!!.forEach { (k, v) ->
+        if (v !is Int || v !is Double) {
+          logger.info { "Deprecated attribute type, please use an Int or Double for $k" }
```
Comment (r3mariano) [inline misk-launchdarkly-core/src/main/kotlin/misk/feature/launchdarkly/LaunchDarklyFeatureFlags.kt:147]: I should really split out misk logging, but for now can you use kotlin-logging directly so we're not pulling in all of core?

**Review by r3mariano — APPROVED** [review]
Looks good!

**Pull request merged**